src/frame/sdk/DocumentApi.ts

```diff
@@ -74,6 +68,57 @@ function normalizeUserAndGroupPublicKeyList(userKeys: UserKeyListResponseType, g
     ];
 }
 
+/**
+ * Get a list of user and group keys for the provided users and then also add in the current user to the list of user keys
+ * @param {string[]} userGrants  List of users to get public keys for
+ * @param {string[]} groupGrants List of groups to get public keys for
+ * @param {boolean}  grantToAuthor If the logged in user should be included in the list of users to encrypt to.
+ * @param {Policy}   policy An optional policy which will be used to lookup users and groups in addition to the explicit userGrants and groupGrants.
+ */
+function getKeyListsForUsersAndGroups(
+    userGrants: string[],
+    groupGrants: string[],
+    grantToAuthor: boolean,
+    policy?: Policy
+): Future<SDKError, {userKeys: UserOrGroupPublicKey[]; groupKeys: UserOrGroupPublicKey[]}> {
+    return Future.gather3(
+        UserApiEndpoints.callUserKeyListApi(userGrants),
+        GroupApiEndpoints.callGroupKeyListApi(groupGrants),
+        PolicyEndpoints.callApplyPolicyApi(policy)
+    ).flatMap(([userKeys, groupKeys, policyResults]) => {
+        if (userKeys.result.length !== userGrants.length || groupKeys.result.length !== groupGrants.length || policyResults.invalidUsersAndGroups.length > 0) {
+            //One of the user or groups in the list here doesn't exist. Fail the create call.
+            const {missingUsersString, missingGroupsString} = createErrorForInvalidUsersOrGroups(
+                userKeys,
+                groupKeys,
+                policyResults.invalidUsersAndGroups,
+                userGrants,
+                groupGrants
+            );
+            return Future.reject(
+                new SDKError(
+                    new Error(
+                        `Failed to create document due to unknown users or groups in access list. Missing user IDs: [${missingUsersString}]. Missing group IDs: [${missingGroupsString}]`
+                    ),
+                    ErrorCodes.DOCUMENT_CREATE_WITH_ACCESS_FAILURE
+                )
+            );
+        } else if (userKeys.result.length === 0 && groupKeys.result.length === 0 && !grantToAuthor && policyResults.usersAndGroups.length === 0) {
```

Comment:
This can just be an `if` since you're returning in the `if` above.

Reply:
ok.